xmtp_mls/src/storage/errors.rs

```diff
@@ -68,6 +68,7 @@ impl RetryableError for StorageError {
             Self::Pool(_) => true,
             Self::Lock(_) => true,
             Self::SqlCipherNotLoaded => true,
+            Self::PoolNeedsConnection => true,
```

Comment:
This means the error will retry 5 times before erroring. That might be okay? I also took a pass at just doing it in the code and retrying 3 times here https://github.com/xmtp/libxmtp/pull/1010/commits/45adbfe9ee1641694db84a3778a11700e16c5e2d open to others thoughts on how we should handle this.

Reply:
This definitely feels like it should be retryable. Thanks for fixing